homeassistant/components/command_line/notify.py

```diff
@@ -44,11 +45,34 @@ def send_message(self, message="", **kwargs):
                 self.command,
                 universal_newlines=True,
                 stdin=subprocess.PIPE,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
                 shell=True,  # nosec # shell by design
             )
-            proc.communicate(input=message, timeout=self._timeout)
+
+            _LOGGER.debug(
+                "Running command: `%s`, with input: %s", self.command, message
```

Comment:
This will also put sensitive data in the logs (if provided on the command line) which we should not.

Reply:
So don't log the message basically?

Reply:
Well, we should not log the actually executed command in this case I would say.

Reply:
Oh, hm. So I basically just borrowed what shell_command and command line sensor were already doing. I can change it to just log stdout and stderr? I suppose the command isn't really necessary it just seemed helpful to the person trying to debug to link the command to the output right in the message. Also its not like this is logged always, only when debugging is specifically turned on.

Anyway I can guess I can just drop this line and log out the entity ID with stdout and stderr, the developer will then have to remember or look up the command. Should it also be removed from shell_command and command line sensor then?

Reply:
You can still log messages? The command seems less important and part of the risk involved.

Reply:
For reference:
https://github.com/home-assistant/core/blob/dev/homeassistant/components/command_line/sensor.py#L179
https://github.com/home-assistant/core/blob/dev/homeassistant/components/shell_command/__init__.py#L100

Reply:
Those should not happen, imho, should not log either. But that is out of scope for this PR.

Reply:
Ok understood. Yea fixing them would be a separate pr I just wanted to confirm that they were bugs. Will adjust the pr to not log the command or message

Reply:
The command is also logged on error by this component (and pretty much every component in this integration). Is that also a problem? [See here](https://github.com/home-assistant/core/blob/0a2bdf20e8e36c5ffdf47235cdf1b4a273ba83ee/homeassistant/components/command_line/notify.py#L59):
```python
if proc.returncode != 0:
  _LOGGER.error("Command failed: %s", self.command)
```

Reply:
Ok updated to no longer log the actual command or message in debug logs. Let me know if the error level messages are a problem (although they are pre-existing so that might be a separate bug).